HARK/ConsumptionSaving/ConsLaborModel.py

```diff
@@ -0,0 +1,567 @@
+        cycles : int
+            Number of times the sequence of periods should be solved.
+        time_flow : boolean
+            Whether time is currently "flowing" forward for this instance.
+        
+        Returns
+        -------
+        None
+        '''
+        params = Params.init_labor_intensive.copy()
+        params.update(kwds)
+        kwds = params
+        
+        IndShockConsumerType.__init__(self,cycles = cycles,time_flow=time_flow,**kwds)
+        self.pseudo_terminal = False
+        self.solveOnePeriod = solveConsLaborIntMarg
+        self.update()
+    
+    def update(self):
+        '''
+        Update the income process, the assets grid, and the terminal solution.
+        
+        Parameters
+        ----------
+        None
+        
+        Returns
+        -------
+        None
+        '''
+        self.updateIncomeProcess()
+        self.updateAssetsGrid()
+        self.updateTranShkGrid()
+         
+    def calcBoundingValues(self):      
+        '''
+        Calculate human wealth plus minimum and maximum MPC in an infinite
+        horizon model with only one period repeated indefinitely.  Store results
+        as attributes of self.  Human wealth is the present discounted value of
+        expected future income after receiving income this period, ignoring mort-
+        ality.  The maximum MPC is the limit of the MPC as m --> mNrmMin.  The
+        minimum MPC is the limit of the MPC as m --> infty.
+        
+        NOT YET IMPLEMENTED FOR THIS CLASS
+        
+        Parameters
+        ----------
+        None
+        
+        Returns
+        -------
+        None
+        '''
+        raise NotImplementedError()
+        
+    def makeEulerErrorFunc(self,mMax=100,approx_inc_dstn=True):
+        '''
+        Creates a "normalized Euler error" function for this instance, mapping
+        from market resources to "consumption error per dollar of consumption."
+        Stores result in attribute eulerErrorFunc as an interpolated function.
+        Has option to use approximate income distribution stored in self.IncomeDstn
+        or to use a (temporary) very dense approximation.
+        
+        NOT YET IMPLEMENTED FOR THIS CLASS
+        
+        Parameters
+        ----------
+        mMax : float
+            Maximum normalized market resources for the Euler error function.
+        approx_inc_dstn : Boolean
+            Indicator for whether to use the approximate discrete income distri-
+            bution stored in self.IncomeDstn[0], or to use a very accurate
+            discrete approximation instead.  When True, uses approximation in
+            IncomeDstn; when False, makes and uses a very dense approximation.
+        
+        Returns
+        -------
+        None
+        '''
+        raise NotImplementedError()
+
+        
+    def getControls(self):
+        '''
+        Calculates consumption for each consumer of this type using the consumption functions.
```

Comment:
I think this docstring is incorrect. There are more controls than just consumption in this model